Update postgres driver and spring versions.

vadeg opened the pull request:
- Update to `spring-4.3.8`
- Update to `postgres-42.1.1`
- Update `pom` version to `2.0.0-SNAPSHOT`
- Fix integration test to support new `postgres` driver version.

valgog commented:
@vadeg do you know, why the travis-ci build failed?

vadeg commented:
@valgog, yes. Because I have added java 8 specific test. Fixed.

vadeg commented:
`java-sproc-wrapper` becomes incompatible with java 7 after `postgresql` driver update. Should I update travis config to exclude java-7 tests?

valgog commented:
@vadeg it would be a ok to drop support for java7 and lower

valgog commented on src/test/java/de/zalando/typemapper/postgres/PgSerializerTest.java:
```diff
@@ -61,8 +58,8 @@ public PgSerializerTest(final Object objectToSerialize, final Object expected) {
         return Arrays.asList(
                 new Object[][] {
-                    {new Date(112, 11, 1, 6, 6, 6), Pattern.compile("2012-12-01 06:06:06.000000 [+-]?\\d{2}:\\d{2}:00")},
-                    {new Date(112, 9, 1, 6, 6, 6), Pattern.compile("2012-10-01 06:06:06.000000 [+-]?\\d{2}:\\d{2}:00")},
```
Do you know why the milliseconds are dropped here? should we add non-zero millisecond containing test?

vadeg replied:
Yes. They have changed `toString` method. Millis will be added if they really exists. See commit [1e1f3c4ab1fbbc5b071b50991da6c34afe92e6f0](https://github.com/pgjdbc/pgjdbc/commit/1e1f3c4ab1fbbc5b071b50991da6c34afe92e6f0)

LappleApple commented:
@valgog @Jan-M @vadeg Checking in on this, is it ready for merge?

vadeg commented:
I have fixed one more exception.
When there is a custom database type which extends `ENUM` is used as an argument for SP and if this argument is null, then Postgres will return exception
```
Caused by: org.postgresql.util.PSQLException: ERROR: function function_abc(character varying) does not exist
  Hint: No function matches the given name and argument types. You might need to add explicit type casts.
```

This happens only when value, which should be passed into PreparedStatement, is `null`. In this case postgres jdbc driver asks database which exact type should be used instead of `java.sql.Types.OTHER`. In case of custom `ENUM` type database will return `VARCHAR`. After that,  postgres will try to lookup function `function_abc` with argument `VARCHAR` instead of `function_abc` with argument `our_custom_type` and will fail.

I have added fix and several tests to check that behaviour. 
This happens because `spring-jdbc` developers slightly changed [code](https://github.com/spring-projects/spring-framework/blob/master/spring-jdbc/src/main/java/org/springframework/jdbc/core/StatementCreatorUtils.java#L240) which determines which type should be used in case of `null` value.
This bug appears in `spring-jdbc-4.3.8` and `spring-jdbc-4.3.9`. According to [spring-jdbc github](https://github.com/spring-projects/spring-framework/blob/master/spring-jdbc/src/main/java/org/springframework/jdbc/core/StatementCreatorUtils.java#L240) there will be no bug in following versions.
